### PR TITLE
Small change to let the possibility to remove or not fully NaNs features

### PR DESCRIPTION
**Pull request by gbrunin**

### CODE REVIEW

**Comment by ppdebreuck** [comment]
Great ! Some questions I have:
1) Why would one use this (what is the issue)? Nans features will never be used by the feature selection ? memory efficiency ?
2) From the implementation, one could either set a custom featurizer (with drop nans to true) when creating the MODData or do It later with featurize() function and dropnan to true. What is the recommended usage ?
3) Should the default behaviour be dropping all nans columns or not ?

**Comment by gbrunin** [comment]
I had to do this because of a quite rare situation I faced. I'm featurizing a composition dataset using, among others, features that are predicted with modnet models. I have trained these models on another database (elastic data materials project) and the features have been selected for this model. In order to featurize my dataset with elastic data predicted from these models, I need to have all the features that have been used to predict the elastic data in my dataset (or featurize twice, but I don't want to do that). From time to time, one of the features that was used for the elastic data model disappears from the features of the dataset because they are all NaNs, which leads to an error when predicting the elastic constants. If I set drop_allnan to False, then these features are still present when predicting the elastic data, and I remove them afterwards myself. 

With this implementation, I have to set drop_allnan to False when initializing the featurizer, and also when featurizing. This could be simplified with e.g. setting self.featurizer.drop_allnan to True if set to True in MODData.featurize() (something similar to set_n_jobs). Do you want me do to that?

The default should be to remove these fully-NaN features, because indeed they do not bring anything. This is just there to help with a peculiar situation for which I have to modify MODNet, otherwise I cannot proceed easily. Thought it might be useful to others as well, and it's not a big change.

File: modnet/models/vanilla.py [hunk 1]
```diff
@@ -846,6 +846,11 @@ def _restore_model(self):
                     fill_value=-1,
                 ).fit(np.zeros((1, self.n_feat))),
             )
+        if not hasattr(self, "targets_groups"):
+            self.targets_groups = [x for subl in self.targets for x in subl]
+            LOG.warning(
+                "Installed modnet version (v>=0.4.0) does not match loaded model (v<0.4.0) and may result in errors. Please retrain or change your modnet version !"
+            )
```
Comment (ml-evs) [inline modnet/models/vanilla.py:853]:
```suggestion
```

I guess we can just leave this for #155 to be merged first

**Review by ppdebreuck — APPROVED** [review]
LGTM 🌞

**Pull request merged**